raiden/transfer/node.py

```diff
@@ -732,156 +747,191 @@ def handle_update_transport_authdata(
         chain_state: ChainState,
         state_change: ActionUpdateTransportAuthData,
 ) -> TransitionResult:
+    assert chain_state is not None, 'chain_state must be set'
     chain_state.last_transport_authdata = state_change.auth_data
     return TransitionResult(chain_state, list())
 
 
-def handle_state_change(chain_state: ChainState, state_change: StateChange) -> TransitionResult:
+def handle_state_change(
+        chain_state: ChainState,
+        state_change: StateChange,
+) -> TransitionResult[ChainState]:
+
     if type(state_change) == Block:
+        assert isinstance(state_change, Block), MYPY_ANNOTATION
```

Comment:
Hmm but why is the assert needed here and in all the files below? Shouldn't the `if type(state_change) == X:` be enough?

Reply:
It should, but it isn't. mypy does not detect it.

Reply:
That's funny. I guess shortcomings of mypy they will hopefully eventually fix. Or ... could it be a subtle different of `type()` vs `isinstance()`? I am just theorizing here. Did not test this.

Reply:
Hmm actually my theory seems to be correct. Let me open a PR for this.

This only throws warning for the `type()` check:

```python
from typing import Union


class Boo(object):
    a = 1


class Foo(object):
    b = 2


def fun(f: Union[Foo, Boo]):
    if type(f) == Foo:
        print(f.b)

    if isinstance(f, Boo):
        print(f.a)
```